src/diffusers/models/cross_attention_processors.py

```diff
@@ -0,0 +1,148 @@
+# Copyright 2022 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import torch
+
+from ..utils.import_utils import is_xformers_available
+
+
+if is_xformers_available():
+    import xformers
+    import xformers.ops
+else:
+    xformers = None
+
+
+class CrossAttentionProcMixin:
+
+    def __init__(self, head_size, upcast_attention):
+        self.head_size = head_size
+        self.upcast_attention = upcast_attention
+
+    def __call__(self, hidden_states, query_proj, key_proj, value_proj, context=None):
+        raise NotImplementedError("Make sure this method is overwritten in the subclass.")
```

Comment:
consider [`@abc.abstractmethod`](https://docs.python.org/3/library/abc.html#abc.abstractmethod) for template methods that must be overridden.